File: services/Users/Services/Users/IntegrationEvents/UserCreatedIntegrationEvent.cs
using System;
using System.Collections.Generic;

namespace Users.Services.Users.IntegrationEvents
{
    public class UserCreatedIntegrationEvent
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public List<string> Roles { get; set; }

        public UserCreatedIntegrationEvent(Guid id, string name, string email, List<string> roles)
        {
            Id = id;
            Name = name;
            Email = email;
            Roles = roles;
        }
    }
}

File: services/learn/learn/application/users/integration_events/user_created/user_created_integration_event.py
import datetime
import uuid
from typing import Any, List

from pydantic import BaseModel

from learn.infrastructure.event_bus.integration.integration_event import IntegrationEvent


class UserCreatedIntegrationEvent(BaseModel, IntegrationEvent):
    id: uuid.UUID
    occurred_on: datetime.datetime
    user_id: uuid.UUID
    name: str
    email: str
    roles: List[str]

    @staticmethod
    def type():
        return "UserCreatedIntegrationEvent"
